som-interpreter-bc/src/block.rs

```diff
@@ -27,7 +27,7 @@ impl Block {
             0 => universe.block1_class(),
             1 => universe.block2_class(),
             2 => universe.block3_class(),
-            _ => panic!("no support for blocks with more than 2 paramters"),
```

Comment:
:) Still busy with SOM-RS? Great to see. I did fix some bits here: https://github.com/SOM-st/som-rs/pull/1

Reply:
Haha, I got spotted ! :)

Yeah, I do plan to come back to it a bit more.  
I did see the fixes you made on your fork, I think I will backport these here as well.  

I also plan to add support for the new bits that were added recently in the SOM core library (like `loadFile:`, `printStackTrace` and others).  

I got the idea some days ago to try doing this year's edition of the Advent of Code using SOM, for fun and to maybe find some more quirks in the interpreter (it could also serve as test cases, possibly).

Reply:
Right now, I am trying to get ReBench back up and running.  

I unfortunately lost my previous ReBench setup and all the previous benchmark data, due to [a fire that destroyed parts of OVH's datacenter in Strasbourg](https://www.reuters.com/article/us-france-ovh-fire-idUSKBN2B20NU), and my server who lived there turned out irrecoverable.

Reply:
> try doing this year's edition of the Advent of Code using SOM

Oh, nice :) Let me know if there's something you need, in the library or so.
The code might also make for interesting new benchmarks :)

>  a fire that destroyed parts of OVH's datacenter 
Ugh, yeah, I heard about that.

For ReBench, there were a few changes since you last touched it, mostly on the master branch, not packaged up as a pip release yet...

Reply:
> For ReBench, there were a few changes since you last touched it, mostly on the master branch, not packaged up as a pip release yet...

Yeah, there was a few more things to do than last time, like the GitHub App integration, but it was still fairly straightforward.
It is now running again at the same URL as before: <https://rebench.polomack.eu>.

> The code might also make for interesting new benchmarks :)

I have now made public the repository where the code for these Advent of Code solutions are available:  
<https://github.com/Hirevo/advent-of-code-2021>

I found the solution for day 4 to already be quite interesting in that regard, it does a lot of work with nested arrays and takes a bit of time to fully execute (~4 seconds on som-java and ~16 seconds on som-rs).  
I am convinced that I'll find some opportunities for optimization after looking at them more closely and when solving more of these puzzles.